### DISCUSSION TITLE
Do we need to recover the module from Bitcode?

### DISCUSSION THREAD

**Discussion by kayaercument** [Munich-Quantum-Software-Stack/QDMI#51] [General]
Last week, I wrote a couple of backends for QPUs at LRZ. I need to translate the QIR code to the QASM code. I was not able to recover the module from the bitcode without the size. Currently, we have yet to get such a request. It would be wise to consider.

FYI @burgholzer @echavarria-lrz

**Comment by burgholzer** [comment]
Maybe a quick question first:
Why do you need to translate QIR to QASM in the backends?
If a backend does not support QIR as an input format, then it should report that it does not support QIR as an input.
IMHO, there should be a central place in the MQSS that handles QIR to QASM translation.
And I also think that we should not force people to implement support for both formats. If they provide both, great. If not, that should be also fine. The software stack should handle the conversion for them.

**Reply by kayaercument** [reply]
What you are suggesting makes sense. How can we determine if the backend supports QIR? IMHO, if `QDMI_control_pack_qir` returns `QDMI_ERROR_NOTIMPL` we can make it work. What do you think?

**Reply by burgholzer** [reply]
That would be my suggestion here, i.e., a backend not supporting QIR should return an error for the respective function calls.
If `QDMI_ERROR_NOTIMPL` is the right error here or whether a different one, e.g. `QDMI_ERROR_NOT_SUPPORTED`, would be more appropriate is a different question.
